pkg/cmd/roachtest/tests/failure_injection.go

```diff
@@ -331,33 +349,247 @@ var latencyTest = func(c cluster.Cluster) failureSmokeTest {
 	}
 }
 
-func setupFailureSmokeTests(ctx context.Context, t test.Test, c cluster.Cluster) error {
+var cgroupsDiskStallTests = func(c cluster.Cluster) []failureSmokeTest {
+	type readWriteBytesOverTime struct {
+		stalledNodeReadBytes     int
+		stalledNodeWriteBytes    int
+		unaffectedNodeReadBytes  int
+		unaffectedNodeWriteBytes int
+	}
+
+	// Returns the read and write bytes read/written to disk over the last 30 seconds of the
+	// stalled node and a control unaffected node.
+	getReadWriteBytesOverTime := func(ctx context.Context, l *logger.Logger, c cluster.Cluster, f *failures.CGroupDiskStaller, stalledNode, unaffectedNode int) (readWriteBytesOverTime, error) {
+		beforeStallReadBytes, beforeStallWriteBytes, err := f.GetReadWriteBytes(ctx, l, c.Node(stalledNode).InstallNodes())
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+		beforeUnaffectedReadBytes, beforeUnaffectedWriteBytes, err := f.GetReadWriteBytes(ctx, l, c.Node(unaffectedNode).InstallNodes())
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+		// Drop the page caches on each VM to force them to read from disk. Without
+		// this, we don't run the workload long enough for the process to read
+		// from disk instead of memory.
+		c.Run(ctx, option.WithNodes(c.CRDBNodes()), "echo 3 | sudo tee /proc/sys/vm/drop_caches")
+		select {
+		case <-ctx.Done():
+			return readWriteBytesOverTime{}, ctx.Err()
+		case <-time.After(30 * time.Second):
+		}
+
+		afterStallReadBytes, afterStallWriteBytes, err := f.GetReadWriteBytes(ctx, l, c.Node(stalledNode).InstallNodes())
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+		afterUnaffectedReadBytes, afterUnaffectedWriteBytes, err := f.GetReadWriteBytes(ctx, l, c.Node(unaffectedNode).InstallNodes())
+		if err != nil {
+			return readWriteBytesOverTime{}, err
+		}
+
+		return readWriteBytesOverTime{
+			stalledNodeReadBytes:     afterStallReadBytes - beforeStallReadBytes,
+			stalledNodeWriteBytes:    afterStallWriteBytes - beforeStallWriteBytes,
+			unaffectedNodeReadBytes:  afterUnaffectedReadBytes - beforeUnaffectedReadBytes,
+			unaffectedNodeWriteBytes: afterUnaffectedWriteBytes - beforeUnaffectedWriteBytes,
+		}, nil
+	}
+
+	var tests []failureSmokeTest
+	for _, stallWrites := range []bool{true, false} {
+		for _, stallReads := range []bool{true, false} {
+			if !stallWrites && !stallReads {
+				continue
+			}
+
+			nodes := c.CRDBNodes()
+			rand.Shuffle(len(nodes), func(i, j int) {
+				nodes[i], nodes[j] = nodes[j], nodes[i]
+			})
+			stalledNode := nodes[0]
+			unaffectedNode := nodes[1]
+
+			testName := fmt.Sprintf("%s: WritesStalled=%t ReadsStalled=%t", failures.CgroupsDiskStallName, stallWrites, stallReads)
```

Comment:
Nit: maybe use `/` as the delimiter in case these later become top-level roachtests.